jwst/outlier_detection/outlier_detection_step.py

```diff
@@ -171,8 +172,23 @@ def process(self, input_data):
 
             state = 'COMPLETE'
             if self.input_container:
+                if not self.save_intermediate_results:
+                    self.log.info("The following files will be deleted since save_intermediate_results=False:")
                 for model in self.input_models:
                     model.meta.cal_step.outlier_detection = state
+                    if not self.save_intermediate_results:
+                        # remove unwanted files
+                        crf_file = self.make_output_path(basepath=model.meta.filename)
+                        outlr_basename = os.path.basename(crf_file)
+                        outlr_file = model.meta.filename.replace('cal', 'outlier_i2d')
+                        outlr_path = crf_file.replace(outlr_basename, outlr_file)
+                        if os.path.isfile(outlr_path):
```

Comment:
Does this statement basically test to see if the file exists before trying to delete it?